server/container/util-java8/src/test/java/org/apache/james/util/streams/JamesCollectorsTest.java

```diff
@@ -0,0 +1,130 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.util.streams;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import com.github.steveash.guavate.Guavate;
+import com.google.common.collect.ImmutableList;
+
+public class JamesCollectorsTest {
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void chunkerShouldAcceptEmptyStrem() {
+        Stream<Integer> emptyStream = Stream.of();
+
+        assertThat(emptyStream.collect(JamesCollectors.chunker(10)))
+            .isEmpty();
+    }
+
+    @Test
+    public void chunkerShouldThrowOnZeroChunkSize() {
+        expectedException.expect(IllegalArgumentException.class);
+
+        JamesCollectors.chunker(0);
+    }
+
+    @Test
+    public void chunkerShouldThrowOnNegativeChunkSize() {
+        expectedException.expect(IllegalArgumentException.class);
+
+        JamesCollectors.chunker(-1);
+    }
+
+    @Test
+    public void chunkerShouldChunkMonoValueStreams() {
+        Stream<Integer> emptyStream = Stream.of(1);
+
+        List<List<Integer>> values = emptyStream.collect(JamesCollectors.chunker(10))
```

Comment:
No, I want to check I have the right structure after chuncking.

What you propose is not equivalent (you don't check chunking, but that I do not loose data)

Akka my test is more powerful...